packages/fxa-settings/src/components/FormPasswordWithBalloons/index.tsx

```diff
@@ -0,0 +1,207 @@
+                      value.toLowerCase()
+                    );
+                  },
+                  uncommon: async (value: string) => {
+                    // @ts-ignore
+                    const list = await import('fxa-common-password-list');
+                    const input = value.toLowerCase();
+                    return (
+                      !list.test(input) && !passwordValidator.isBanned(input)
+                    );
+                  },
+                },
+              })}
+              prefixDataTestId="new-password"
+            />
+          </FtlMsg>
+          {isNewPwdBalloonVisible && (
+            <PasswordStrengthBalloon
+              {...{
+                hasUserTakenAction: formState.dirtyFields.newPassword,
+                isTooShort: errors.newPassword?.types?.length,
+                isSameAsEmail: errors.newPassword?.types?.notEmail,
+                isCommon: errors.newPassword?.types?.uncommon,
+              }}
+            />
+          )}
+        </div>
+
+        <div className="relative">
+          <FtlMsg
+            id="form-reset-password-with-balloon-confirm-password"
+            attrs={{ label: true }}
+          >
+            <InputPassword
+              name="confirmPassword"
+              label={confirmPasswordLabel || 'Re-enter password'}
+              className="text-start"
+              onFocusCb={showConfirmPwdBalloon ? onConfirmPwdFocus : undefined}
+              onBlurCb={
+                showConfirmPwdBalloon && isConfirmPwdBalloonVisible
+                  ? onConfirmPwdBlur
+                  : undefined
+              }
+              onChange={() => {
+                if (passwordMatchErrorText) {
+                  setPasswordMatchErrorText(undefined);
+                }
+                trigger(['newPassword', 'confirmPassword']);
+              }}
+              // Password confirmation field is disabled until new password is valid
+              disabled={
+                !formState.dirtyFields.newPassword || errors.newPassword
+              }
+              hasErrors={errors.confirmPassword && passwordMatchErrorText}
+              errorText={passwordMatchErrorText}
+              inputRef={register({
+                required: true,
+                validate: (value: string) => value === getValues().newPassword,
+              })}
+              anchorStart
+              tooltipPosition="bottom"
+              prefixDataTestId="verify-password"
+            />
+          </FtlMsg>
+
+          {showConfirmPwdBalloon && isConfirmPwdBalloonVisible && (
+            <div className="input-balloon flex gap-2 items-start text-xs">
+              <IconKey className="w-10" />
+              <p>
+                You need this password to access any encrypted data you store
+                with us.
+                <br className="mb-3" />A reset means potentially losing data
+                like passwords and bookmarks.
+              </p>
```

Comment:
Q: Does this need to be wrapped in an FtlMsg block?

Reply:
Yes it does! I'll add that.